hamster_gtk/tracking/screens.py

```diff
@@ -238,3 +268,119 @@ def _on_start_tracking_button(self, button):
+        self.show_all()
+
+
+class RecentActivitiesGrid(Gtk.Grid):
+    """A widget that lists recent activities and allows for quick continued tracking."""
+
+    def __init__(self, start_tracking_widget, controller, *args, **kwargs):
+        """
+        Initiate widget.
+
+        Args:
+            start_tracking_widget (StartTrackingBox): Is needed in order to set the raw fact.
+            controller: Is needed in order to query for recent activities.
+        """
+        super(Gtk.Grid, self).__init__(*args, **kwargs)
+        self._start_tracking_widget = start_tracking_widget
+        self._controller = controller
+
+        self._controller.signal_handler.connect('facts-changed', self.refresh)
+        self._populate()
+
+    def refresh(self, sender=None):
+        """Clear the current content and re-populate and re-draw the widget."""
+        helpers.clear_children(self)
+        self._populate()
+        self.show_all()
+
+    def _populate(self):
+        """Fill the widget with rows per activity."""
+        def add_row_widgets(row_counter, activity):
+            """
+            Add a set of widgets to a specific row based on the activity passed.
+
+            Args:
+                row_counter (int): Which row to add to.
+                activity (hamster_lib.Activity): The activity that is represented by this row.
+            """
+            def get_label(activity):
+                """Label representing the activity/category combination."""
+                label = Gtk.Label(helpers.serialize_activity(activity))
+                label.set_halign(Gtk.Align.START)
+                return label
+
+            def get_copy_button(activity):
+                """
+                A button that will copy the activity/category string to the raw fact entry.
+
+                The main use case for this is a user that want to add a description or tag before
+                actually starting the tracking.
+                """
+                button = Gtk.Button('Copy')
+                activity = helpers.serialize_activity(activity)
+                button.connect('clicked', self._on_copy_button, activity)
+                return button
+
+            def get_start_button(activity):
+                """A button that will start a new ongoing fact based on that activity."""
+                button = Gtk.Button('Start')
+                activity = helpers.serialize_activity(activity)
+                button.connect('clicked', self._on_start_button, activity)
+                return button
+
+            self.attach(get_label(activity), 0, row_counter, 1, 1)
+            self.attach(get_copy_button(activity), 1, row_counter, 1, 1)
+            self.attach(get_start_button(activity), 2, row_counter, 1, 1)
+
+        today = datetime.date.today()
+        start = today - datetime.timedelta(90)
+        activities = reversed(helpers.get_recent_activities(self._controller, start, today))
```

Comment:
Do we really want all the activities and scroll instead of displaying the `tracking_recent_activities_items`?

Reply:
I would prefer such a more general solution. There is nothing prefenting us from limiting the results to just ``tracking_recent_activities_items`` (via config) if we want, but especially given that we already have a solution I do not see the harm. 

We **need** scrolling anyway (because we can not make any guarantee that ``tracking_recent_activities_items`` will fit in the space available. In light of that, it seems we get the extra activities for free if the user bothers scolling.

What would be the downside?

Reply:
Accidental scrolling to the other end and performance.

Reply:
- "accidental scrolling": Im not sure I see the use case. especially as there is the posibility of a scrollbox even if if we just fetch the number of activities defined in the config.
- by default we just fetch todays facts. this should net us no more than a couple of dozens facts at most I figure. I doubt there is any concievable performance impact on the DB or the GUI rendering. Unless I'm missing something this seems like a case of premature optimization.

Reply:
* You scroll somewhere accidentally and since the list is long you do not have a simple way of returning back – you cannot just quickly scroll down and when you find out that you are on the wrong side quickly scroll to the correct one.
* I see more than just today’s facts. In fact there does not seem to be any limit.

Reply:
You are [right](https://github.com/projecthamster/hamster-gtk/blob/feature/136/hamster_gtk/tracking/screens.py#L354)!

The idea was to make the timedelta adjustable via a config setting, but i forgot to actually do that :(
If done that way it is up to the user to figure what timeframe is viable to them. We probably should point out the performance consideration though...